backend/routes/organization.js

```diff
@@ -1,3 +1,290 @@
-/*
-    This file should one day contain routes related to organizations.
-*/
\ No newline at end of file
+// npm Packages
+const express = require('express');;
+const bodyParser = require('body-parser');
+const router = express.Router();
+
+// Local imports
+const Organization = require('mongoose').model('Organization');
+
+router.get('/test', (req, res) => {
+    res.send('test successful');
+});
+
+//For all organizations
+router.post('/', (req,res) => {
+    console.log(req.body);
+    Organization.create(req.body, (err, organization) => {
+        if (err) {
+            res.send("" + err);
+        } else {
+            console.log("Created Organization");
+            res.send(organization);
+        }
+    });
+});
+
+router.get('/', (req, res) => {
+    console.log("Getting all Organizations");
+    Organization.find({})
+        .exec().then((organization) => res.send(organization))
+        .catch((err) => {
+            res.send("" + err);
+        });
+});
+
+//For specific organization
+router.get('/:organization_id', (req, res) => {
+    Organization.findById(req.params.organization_id).then((organization) => res.send(organization))
+      .catch((err) => res.send("" + err));
+});
+
+router.put('/:organization_id', (req, res) => {
+    Organization.findByIdAndUpdate(req.params.organization_id, req.body, {upsert: true, new: true}, function(err, response) {
+        if (err) {
+            res.json("" + err);
+        } else {
+            res.json(response);
```

Comment:
I wouldn't send back the whole object on these put / post calls, just a 200 OK with the ID of any newly created resource.

We can update it later but it's somewhat redundant to return to the client exactly what they just sent you. (I may have missed that in the mentors api.)